### ISSUE TITLE
Warn on security_groups and subnet_id usage on aws_instance

### ISSUE THREAD

**Issue by clstokes** [hashicorp/terraform#6638]
### tldr;

Can we add a warning for usage of `security_groups` on `aws_instance` resources that also specify `subnet_id`? This causes challenges for users of VPCs; challenges that are not immediately obvious.
### Details

EC2-Classic instances are unable to have their security groups changed, so when `security_groups` on an `aws_instance` is changed, Terraform correctly forces the instance to be recreated. For instances in a VPC, `vpc_security_group_ids` is used so that security groups can be changed on an existing instance. It's not always clear to users the implications of using `security_groups` vs `vpc_security_group_ids` so they can get in a tough spot if they need to change security groups on an instance that was created in a VPC but using `security_groups`.  A warning would help users avoid this tough spot.
### Terraform Version

0.6.16 and all prior versions.
### Affected Resource(s)
- aws_instance
### Terraform Configuration Files

``` hcl
resource "aws_instance" "bastion" {
  ami                  = "ami-9a562df2"
  security_groups      = ["${aws_security_group.bastion.id}"]
  subnet_id            = "${var.subnet_id}"
  instance_type        = "t2.small"
}

resource "aws_security_group" "bastion" {
  name        = "bastion"
  description = "bastion"
  vpc_id = "${var.vpc_id}"
}

variable "vpc_id" {}
variable "subnet_id" {}
```

**Comment by clstokes** [comment]
It seems a warning may have existed at one point?  https://github.com/hashicorp/terraform/issues/6369#issuecomment-215091170
